sphinx_gallery/backreferences.py

```diff
@@ -227,3 +232,21 @@ def write_backreferences(seen_backrefs, gallery_conf,
             ex_file.write(_thumbnail_div(build_target_dir, fname, snippet,
                                          is_backref=True))
             seen_backrefs.add(backref)
+
+
+def finalize_backreferences(seen_backrefs, gallery_conf):
+    """Replace backref files only if necessary."""
+    logger = sphinx_compatibility.getLogger('sphinx-gallery')
+    if gallery_conf['backreferences_dir'] is None:
+        return
+
+    for backref in seen_backrefs:
+        path = os.path.join(gallery_conf['src_dir'],
+                            gallery_conf['backreferences_dir'],
+                            '%s.examples.new' % backref)
+        if os.path.isfile(path):
+            _replace_md5(path)
+        else:
+            logger.warning('Could not find backreferences file: %s' % (path,))
+            logger.warning('The backreferences are likely to be erroneous '
```

Comment:
This warning stops compilation if compiled with `-W`.

Reply:
The `warning` level seems appropriate for most cases. We could add a new `config` option to allow you to turn down the `backreferences` warning level for repos that really want to do so for people who have code that requires case-sensitivity building on file systems that are case insensitive.

Reply:
Well, lets see how it works in the wild...